Dev

HelloMosquito opened the pull request:
1. completed all necessary components in messenger's chat page
2. refactored code
3. directory structure needs to be rebuilt after completing other tickets

HelloMosquito committed:
85% finished signup

HelloMosquito committed:
1. complete Ticket-1 features; 2. code needs to be rebuilt; 3. squashed commits

HelloMosquito commented:
1. squashed commits and re-push to dev-branch
2. complete Ticket-1 features
3. removed the user's profile photo in chat as discussed in slash

sina-jamshidi commented on client/src/components/ChatAndSearchbar.Component.jsx:
```diff
@@ -0,0 +1,29 @@
+            <Box className={classes.chatAndSearchbarContainer} >
+                <Box className={classes.contactListChatTitle}>Chats</Box>
+                <SearchbarComponent />
+            </Box>
```
A recommendation is to use `<Grid>` instead of `<Box>`. Grids are wrappers around Box that include flexbox automatically so you need way less styling

sina-jamshidi commented on client/src/components/MsgBubbleFromOthers.Component.jsx:
```diff
@@ -0,0 +1,73 @@
+						<Box className={classes.msgBubble}>aoijf fsadfasdfoiwaefjoiwai aoijfoidfasdfwaefjoiwai aoijfoiwafasdfefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjsdfdsfoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai</Box>
+					</Box>
+				</Box>
+			</Box> */}
```
remove commented code from PRs

sina-jamshidi commented on client/src/components/MsgBubbleFromUser.Component.jsx:
```diff
@@ -0,0 +1,59 @@
+						<Box className={classes.msgBubble}>aoijf fsadfasdfoiwaefjoiwai aoijfoidfasdfwaefjoiwai aoijfoiwafasdfefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjsdfdsfoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai aoijfoiwaefjoiwai</Box>
+					</Box>
+				</Box>
+				{/* <Avatar className={classes.avatar} alt={"user"} src="/images/thomas.png" /> */}
```
remove commented code

sina-jamshidi commented on client/src/components/MsgTyping.Component.jsx:
```diff
@@ -0,0 +1,68 @@
+						)
+					}}
+				/>
+			</Box> */}
```
remove commented code

sina-jamshidi commented on client/src/components/MsgTyping.Component.jsx:
```diff
@@ -0,0 +1,68 @@
+				InputProps={{
+					disableUnderline: true,
+					endAdornment: (
+						<InputAdornment>
```
Not sure if this component is necessary. You could just wrap the other components in Fragments. I might be wrong though

sina-jamshidi commented on client/src/components/Searchbar.Component.jsx:
```diff
@@ -0,0 +1,40 @@
+				disableUnderline: true,
+				borderRadius: 50,
+				startAdornment: (
+					<InputAdornment>
```
same note here on input adornment

sina-jamshidi commented on client/src/pages/Messenger.jsx:
```diff
@@ -0,0 +1,75 @@
+		minHeight: "100vh",
+		// "& .MuiInput-underline:before": {
+		// 	borderBottom: "1.2px solid rgba(0, 0, 0, 0.2)"
+		// }
```
remove commented code

HelloMosquito committed:
Merge branch 'test' into temp

sina-jamshidi reviewed (commented):
This looks good overall! take a look into using a formatter like `prettier` for more consistent code style.
- Responsiveness-wise it's nice that the left bar disappears when the screen is too small, but it would be good to be able to access it again with a button.
- I noticed that the chat portion does not automatically slide to the bottom
- While you added end adornments to the textbox, they are not clickable buttons
- There were multiple warnings when running the client
![wenMessenger](https://user-images.githubusercontent.com/52902272/109891821-9c7a2800-7c3e-11eb-9b00-b9777698b545.png)

HelloMosquito committed:
1. complete majority features of Ticket 3; 2. have not start the postgresql init sql; 3. have not start modify readme

HelloMosquito commented:
1. Mainly write the backend ('server' directory), not too much modification in the 'client' directory.
2. Have not written the readme file
3. have not written the db init sql
4. complete the majority of features in Ticket 3 about the backend

HelloMosquito committed:
1. complete db init sql script; 2. modified readme; 3. fixed login bug

HelloMosquito committed:
I - 1. complete majority of Ticket2; 2. added signup.spec.js unit test; 3. fixed some bugs issued in Ticket3

II - refacotr the messenger page; 2. auto scroll to bottom when loading

III - fixed unrecognized props bug in FE

IV - complete auto scale of the msg type input

V - cleaned code

HelloMosquito committed:
Ticket3 (#8)

* 85% finished signup

* 1. complete majority features of Ticket 3; 2. have not start the postgresql init sql; 3. have not start modify readme

* 1. complete db init sql script; 2. modified readme; 3. fixed login bug

HelloMosquito committed:
Revert "Ticket3 (#8)"

This reverts commit cf9b558adbc7dc134bce1380ad1fbb1fa8865b35.

HelloMosquito committed:
Merge branch 'dev' into Ticket3

HelloMosquito committed:
fixed some bugs

tmattmac commented on server/services/login.service.js:
```diff
@@ -7,12 +8,27 @@ const loginService = async (email, password) => {
     let user = await findUserByEmail(email);
+    let loginResponse;
+    if (!user) {
+      loginResponse = {
+        loginStatus: false,
+        msg: "User doesn't exist.",
+      };
+      return loginResponse;
+    }
+    if (!bcrypt.compareSync(password, user.passwd)) {
+      loginResponse = {
+        loginStatus: false,
+        msg: "Invalid password, please try again.",
+      };
```
When logging in, instead of showing a specific message for username and password, it would be better to show a more general message to prevent attackers from probing for existing usernames.

sina-jamshidi commented on server/config/database.config.js:
```diff
@@ -0,0 +1,10 @@
+const dbConfiguration = {
+  dev: {
+    username: "postgres",
+    password: "1002",
```
this hsould not be public information
